### PR TITLE
Fracture integral changes needed for XFEM

### LINKED ISSUE #6392: Fracture integral improvements needed for XFEM
### Description of the enhancement or error report

- Enable use of ```intersecting_boundaries``` when a crack is defined with a set of points rather than nodes.
- Update q functions when the mesh changes 

### Rationale for the enhancement or information for reproducing the error

The fracture integrals aren't correct for meshes cut with XFEM because they depend on an AuxKernel that is currently only run at initialize, and not after the mesh changes (as it does for XFEM).

### PR DESCRIPTION
**Pull request by bwspenc**
closes #6392

Relevant test cases are included in XFEM pull request #6321

### CODE REVIEW

**Comment by bwspenc** [comment]
@jiangwen or @backmari Could one of you please review this?

**Comment by bwspenc** [comment]
make that @jiangwen84

File: modules/solid_mechanics/src/userobjects/CrackFrontDefinition.C [hunk 1]
```diff
@@ -112,8 +112,6 @@ CrackFrontDefinition::CrackFrontDefinition(const InputParameters & parameters) :
   if (isParamValid("intersecting_boundary"))
   {
     _intersecting_boundary_names = getParam<std::vector<BoundaryName> >("intersecting_boundary");
-    if (_geom_definition_method == CRACK_FRONT_POINTS)
-      mooseError("The use of 'intersecting_boundary' together with 'crack_front_points' is not yet supported");
   }
```
Comment (jiangwen84) [inline modules/solid_mechanics/src/userobjects/CrackFrontDefinition.C]: we need to remove the ```{ }```

File: modules/solid_mechanics/src/userobjects/CrackFrontDefinition.C [hunk 2]
```diff
@@ -1183,7 +1181,11 @@ CrackFrontDefinition::isPointWithIndexOnIntersectingBoundary(const unsigned int
...
   else
   {
-    //TODO: Implement for CRACK_FRONT_POINTS
+    // If the intersecting boundary option is used with crack front points, the
+    // first and last points are assumed to be on the intersecting boundaries.
+    unsigned int num_crack_front_points = getNumCrackFrontPoints();
+    if (point_index == 0 || point_index == num_crack_front_points - 1)
+      is_on_boundary = true;
   }
```
Comment (jiangwen84) [inline modules/solid_mechanics/src/userobjects/CrackFrontDefinition.C]: What happen if the points form a closed loop? Would that be an issue?

File: modules/solid_mechanics/src/userobjects/CrackFrontDefinition.C (hunk 1 above)
Comment (bwspenc) [inline modules/solid_mechanics/src/userobjects/CrackFrontDefinition.C]: Good call. I'll fix it.

File: modules/solid_mechanics/src/userobjects/CrackFrontDefinition.C (hunk 2 above)
Comment (bwspenc) [inline modules/solid_mechanics/src/userobjects/CrackFrontDefinition.C]: Another good call.  We should disallow the use of intersecting_boundary if it's a closed loop.  I'll do that.

Comment (bwspenc) [inline modules/solid_mechanics/src/userobjects/CrackFrontDefinition.C]: The new version I just pushed adds a check for this.

Comment (jiangwen84) [inline modules/solid_mechanics/src/userobjects/CrackFrontDefinition.C]: :+1:

Comment (bwspenc) [inline modules/solid_mechanics/src/userobjects/CrackFrontDefinition.C]: I just slightly revised it.  Same idea, but no need to get the info from the parameters the way I did.

Comment (jiangwen84) [inline modules/solid_mechanics/src/userobjects/CrackFrontDefinition.C]: I see. Looks good to me! :+1:

**Comment by permcody** [comment]
:+1:

**Pull request merged**